src/test/java/edu/harvard/iq/dataverse/ingest/metadataextraction/impl/plugins/netcdf/NetcdfFileMetadataExtractorTest.java

```diff
@@ -0,0 +1,62 @@
+package edu.harvard.iq.dataverse.ingest.metadataextraction.impl.plugins.netcdf;
+
+import edu.harvard.iq.dataverse.DatasetFieldConstant;
+import edu.harvard.iq.dataverse.ingest.metadataextraction.FileMetadataIngest;
+import java.io.File;
+import java.util.Map;
+import java.util.Set;
+import static org.junit.Assert.assertNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import org.junit.jupiter.api.Test;
+
+public class NetcdfFileMetadataExtractorTest {
+
+    /**
+     * Expect some lat/long values (geospatial bounding box) with longtude
+     * values that have been transformed from a domain of 0 to 360 to a domain
+     * of -180 to 180.
+     */
+    @Test
+    public void testExtractLatLong() throws Exception {
+        String pathAndFile = "src/test/resources/netcdf/ICOADS_R3.0.0_1662-10.nc";
+        File file = new File(pathAndFile);
+        NetcdfFileMetadataExtractor instance = new NetcdfFileMetadataExtractor();
+        FileMetadataIngest netcdfMetadata = instance.ingestFile(file);
+        Map<String, Set<String>> map = netcdfMetadata.getMetadataMap();
+        assertEquals("-16.320007", map.get(DatasetFieldConstant.westLongitude).toArray()[0]);
+        assertEquals("-6.220001", map.get(DatasetFieldConstant.eastLongitude).toArray()[0]);
+        assertEquals("41.8", map.get(DatasetFieldConstant.southLatitude).toArray()[0]);
+        assertEquals("49.62", map.get(DatasetFieldConstant.northLatitude).toArray()[0]);
+    }
+
+    /**
+     * The NetCDF file under test doesn't have values for latitude/longitude
+     * (geospatial bounding box).
+     */
+    @Test
+    public void testExtractNoLatLong() throws Exception {
+        String pathAndFile = "src/test/resources/netcdf/madis-raob";
+        File file = new File(pathAndFile);
+        NetcdfFileMetadataExtractor instance = new NetcdfFileMetadataExtractor();
+        FileMetadataIngest netcdfMetadata = null;
+        netcdfMetadata = instance.ingestFile(file);
+        Map<String, Set<String>> map = netcdfMetadata.getMetadataMap();
+        assertNull(map.get(DatasetFieldConstant.westLongitude).toArray()[0]);
+        assertNull(map.get(DatasetFieldConstant.eastLongitude).toArray()[0]);
+        assertNull(map.get(DatasetFieldConstant.southLatitude).toArray()[0]);
+        assertNull(map.get(DatasetFieldConstant.northLatitude).toArray()[0]);
+    }
+
+    @Test
+    public void testStandardLongitude() {
+        NetcdfFileMetadataExtractor extractor = new NetcdfFileMetadataExtractor();
+        // Both are over 180. Change.
+        assertEquals(new WestAndEastLongitude("-16.320007", "-6.220001"), extractor.getStandardLongitude(new WestAndEastLongitude("343.68", "353.78")));
+        // One is over 180. Change
+        assertEquals(new WestAndEastLongitude("-260.0", "-179.0"), extractor.getStandardLongitude(new WestAndEastLongitude("100", "181")));
+        // Both are under 180. No change.
```

Comment:
Yes, I agree. I'll ping Jan for some input on this.

Reply:
@pdurbin just for clarification, the right hand-side extractor transforms by subtracting 360° from both?

If so, then I have to agree, this test should fail and truly evaluate to $[100°, -179°]$ due to the above [comment](https://github.com/IQSS/dataverse/pull/9541#discussion_r1185691536) and that there can't be -260° since the domain has it's lower bound at -180°. 

Given the proposed change in the calculation, the test should look like this:

```java
assertEquals(
    new WestAndEastLongitude("100.0", "-179.0"),
    extractor.getStandardLongitude(new WestAndEastLongitude("100", "181"))
);
```

What do you think?

Reply:
I think it's late on a Friday but I powered through and pushed some fixes around longitude at 2edb351.

Almost as importantly, I added docs! Rules in the User Guide about how users (and QA) should expect the code to behave. I'd love to hear what you (@JR-1991) and @atrisovic and @sekmiller (whom I've been bugging all day) think! Here they are:

- West Longitude and East Longitude are expected to be in the range of -180 and 180. (When using :ref:`geospatial-search`, you should use this range for longitude.)
- If West Longitude and East Longitude are both over 180 (outside the expected -180:180 range), 360 will be subtracted to shift the values from the 0:360 range to the expected -180:180 range.
- If either West Longitude or East Longitude are less than zero but the other latitude is greater than 180 (which would imply an indeterminate domain, a lack of clarity of if the domain is -180:180 or 0:360), metadata will be not be extracted.

I also added more tests. I'm happy to add more if you all think of more edge cases. I don't have real NetCDF files to test with that have wacky values. It might be a challenge for QA to find NetCDF files to exercise all the rules.